### PR TITLE
Upgrade ruby to 3.2.2, replace lunr.js with algolia search

### PR DESCRIPTION
**Pull request by noahp**
- **Reapply "Update ruby (#486)" (#490)**
- **algolia**
- **temporarily use the index for docs.memfault.com for testing, until we get the interrupt index set up**

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -49,62 +38,60 @@ GEM
...
+    strscan (3.1.0)
     terminal-table (2.0.0)
       unicode-display_width (~> 1.1, >= 1.1.1)
     unicode-display_width (1.8.0)
-    webrick (1.7.0)
+    webrick (1.8.1)
 
 PLATFORMS
-  ruby
+  x86_64-linux-gnu
```
Comment (gminn) [inline Gemfile.lock:83]: Is there a reason you added the linux platform? Seeing that there may be a concern with hosting services requiring the Linux platform to be specified https://www.moncefbelyamani.com/understanding-the-gemfile-lock-file/#platforms. Wondering if you ran into this?

File: _includes/script.html [hunk 2]
```diff
@@ -4,19 +4,13 @@
...
-        emptyMsg  : '<p>Nothing found.</p>'     // shown message if search returns no results
-      });
-    });
+<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@docsearch/css@3" />
+<script src="https://cdn.jsdelivr.net/npm/@docsearch/js@3" ></script>
+<script>
+docsearch({
+  container: '#docsearch',
+  appId: '4SXF6GYFKZ',
+  apiKey: '2028276a4fc7f09eeb0bd082fbd85164',
```
Comment (gminn) [inline _includes/script.html:13]: Should we rotate this and put it in a github secret, or is it not very sensitive?

**Review by gminn — CHANGES_REQUESTED** [review]
A couple questions! The preview looks great 🤩 🔍

File: Gemfile.lock (hunk 1 above)
Comment (noahp) [reply Gemfile.lock:83]: Whoops, this is an error- my bundler generates platform-specific lockfile by default, need to fix it!

File: _includes/script.html (hunk 2 above)
Comment (noahp) [reply _includes/script.html:13]: It's exposed in the frontend code in the generated site, not considered secret- it's a read-only key.

**Review by gminn — APPROVED** [review]
Looks great! All good to go once we get the interrupt index

**Comment by noahp** [comment]
#528 replaces this, closing

**Pull request closed without merging**